onnx/shape_inference/shape_inference.h

```diff
@@ -0,0 +1,95 @@
+#pragma once
+
+#include "onnx/proto_utils.h"
+#include "onnx/defs/schema.h"
+
+namespace ONNX_NAMESPACE { namespace shape_inference {
+
+struct InferenceContextImpl : public InferenceContext {
+  virtual const AttributeProto* getAttribute(std::string name) {
+    auto attr = attributesByName_.find(name);
+    if (attr == attributesByName_.end()) {
+      return nullptr;
+    } else {
+      return &attr->second;
+    }
+  }
+  virtual const std::vector<TypeProto_Tensor>& getAllInputTypes() {
+    return allInputTypes_;
+  }
+  virtual Status setAllOutputTypes(std::vector<TypeProto_Tensor>&& allOutputTypes) {
+    allOutputTypes_ = allOutputTypes;
+    return Status::Success;
+  }
+  std::unordered_map<std::string, AttributeProto> attributesByName_;
+  std::vector<TypeProto_Tensor> allInputTypes_;
+  std::vector<TypeProto_Tensor> allOutputTypes_;
+};
+
+void InferShapes(ONNX_NAMESPACE::ModelProto& m) {
+  auto g = m.mutable_graph();
+
+  std::unordered_map<std::string, TypeProto_Tensor> valueTypesByName;
+  {
```

Comment:
Why the braces here?

Reply:
it's just organizational, and in some similar places (though not this one) it makes it clear that any temporary variables inside are in fact temporary.

I could take it out if you want me to